File: cpp/02 Constructor , Copy Constructor and Case Study/java/Employee.java
public class Employee{
 
	private int employeeId;
	private String employeeName;
	private String religion;
	public Employee(){
	   	System.out.println("Calling Default Constructor");
   		employeeId = 0;
   		employeeName = "";
   		religion = "Islam";
	   }
	   public Employee(int employeeId){
	   	System.out.println("Calling param constructor with Id");
   		this.employeeId = employeeId;
   		employeeName = "";
   		religion = "Islam";
	   } 
	public void setEmployeeId(int employeeId){
		this.employeeId = employeeId;
	}
	public int getEmployeeId(){
		return this.employeeId;
	}
	public void setEmployeeName(String employeeName){
	 	this.employeeName = employeeName;
	 }
	 public String getEmployeeName(){
	 	return this.employeeName;
	 }
	 void setReligion(String religion){
	   this.religion = religion;
	   
	   
	   }
	String getReligion(){
		return this.religion;
	}
}

File: cpp/02 Constructor , Copy Constructor and Case Study/CPP/CPP/Employee.cpp
#include <iostream>
using namespace std;
class Employee{
    private:
      int employeeId;
      char* employeeName;
      char * religion;
   public :
   	
	   Employee(){
	   	cout<<"Calling Default Constructor"<<endl;
   		employeeId = 0;
   		employeeName = "";
   		religion = "Islam";
	   }
	   Employee(int employeeId){
	   	cout<<"Calling param constructor with Id"<<endl;
   		this->employeeId = employeeId;
   		employeeName = "";
   		religion = "Islam";
	   }   
   	void setReligion(char* religion){
	   this->religion = religion;
	   
	   
	   }
	char* getReligion(){
		return this->religion;
	}
   	void setEmployeeId(int employeeId){ 
   			this->employeeId = employeeId;
	   }
	 int  getEmployeeId(){
	 	return this->employeeId;
	 }
	 void setEmployeeName(char* employeeName){
	 	this->employeeName = employeeName;
	 }
	 char* getEmployeeName(){
	 	return this->employeeName;
	 }
	 
};

